cinema/views.py

```diff
@@ -127,6 +126,54 @@ def upload_image(self, request, pk=None):
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This is a code quality issue. The `upload_image` method is doing too much. It's not only handling the request but also validating the serializer and saving the instance. It would be better to move the validation and saving logic to the serializer to keep the view cleaner and more focused on handling the request.